src/serializer/ResidualHeapVisitor.js

```diff
@@ -526,28 +559,22 @@ export class ResidualHeapVisitor {
           }: ResidualFunctionBinding)
       );
     } else {
+      invariant(environment instanceof DeclarativeEnvironmentRecord);
       // DeclarativeEnvironmentRecord binding
-      invariant(!Environment.IsUnresolvableReference(this.realm, reference));
-      let referencedBase = reference.base;
-      let referencedName: string = (reference.referencedName: any);
-      if (typeof referencedName !== "string") {
-        throw new FatalError("TODO: do not know how to visit reference with symbol");
-      }
-      invariant(referencedBase instanceof DeclarativeEnvironmentRecord);
       let residualFunctionBindings = getOrDefault(
         this.declarativeEnvironmentRecordsBindings,
-        referencedBase,
+        environment,
         () => new Map()
       );
-      let createdBinding = !residualFunctionBindings.has(referencedName);
-      residualFunctionBinding = getFromMap(residualFunctionBindings, referencedName, (): ResidualFunctionBinding => {
-        invariant(referencedBase instanceof DeclarativeEnvironmentRecord);
-        let binding = referencedBase.bindings[referencedName];
+      let createdBinding = !residualFunctionBindings.has(name);
+      residualFunctionBinding = getFromMap(residualFunctionBindings, name, (): ResidualFunctionBinding => {
+        invariant(environment instanceof DeclarativeEnvironmentRecord);
+        let binding = environment.bindings[name];
```

Comment:
This might be a good place to assert that binding will always be defined. After poking through the call site code it seems to me that this is a precondition. It would be nice if I did not have to spend so much effort to find it out.